src/bitutils.h

```diff
@@ -1,14 +1,21 @@
 #pragma once
 #include <cstdint>
 #include <cstdlib>
+#include <cstring>
+#include <type_traits>
 
 // reinterpret_cast for value types
```

Comment:
The comment is still correct, as it just describes what the function does: bit_cast is the effectively the equivalent of reinterpret_cast but for value types.

reinterpret_cast lets you cast pointers:
reinterpret_cast<uint32_t*>(float* a) would cast a float pointer to a u32 pointer, if you dereference this pointer you end up with the same functionality as bit_cast

bit_cast lets you cast the values:
bit_cast<uint32_t>(float a) would cast a float to a uint32_t without type conversion, so a float 1.0f would become uint32_t 0x3F800000